.github/actions/setup/action.yml

```
@@ -0,0 +1,44 @@
+name: 'Setup Java and Dependency Cache'
+description: "Configures the build environment and caches Gradle and dependencies."
+runs:
+  using: "composite"
+  steps:
+    - name: Set Env
+      shell: bash
+      run: |
+        echo "home=${HOME}" >> $GITHUB_ENV
```

```suggestion
        echo "home=${HOME}" >> "$GITHUB_ENV"
```

(In case `$GITHUB_ENV` contains a space)